src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -665,14 +677,48 @@ public virtual string CreateETag(ResourceContext resourceContext)
 
             foreach (IEdmStructuralProperty complexProperty in complexProperties)
             {
+                if (selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes.ContainsKey(complexProperty))
+                {
+                    // We will write the property when the property is expanded. 
+                    continue;
+                }
+
+                ODataNestedResourceInfo nestedResourceInfo = new ODataNestedResourceInfo
+                {
+                    IsCollection = complexProperty.Type.IsCollection(),
+                    Name = complexProperty.Name
+                };
+
+                writer.WriteStart(nestedResourceInfo);
+                WriteComplexAndExpandedNavigationProperty(complexProperty, null, resourceContext, writer, null);
+                writer.WriteEnd();
+            }
+
+            foreach (var item in selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes)
+            {
+                IEdmProperty complexProperty = item.Key;
+                IList<ExpandedReferenceSelectItem> selectItems = null;
+                if (selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes != null)
+                {
+                    if (selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes.ContainsKey(complexProperty))
+                    {
+                        selectItems = selectExpandNode.ExpandedNavigationPropertiesOnComplexTypes[complexProperty];
+                    }
+                }
+
+                if (!resourceContext.StructuredType.DeclaredProperties.Contains(complexProperty))
```

Comment:
>DeclaredProperties [](start = 52, length = 18)

should use `Properties()`, otherwise the complexProperty on the baseType can't be found

Reply:
Add a test case to verify this scenario:

complexType (BaseType)
{
    Property1 (complexType)
}

ComplexType(SubType)
{
   Property2 (complexType)
}

EntityType
{
    Property (SubType)
}

~/....$expand=Property/Property1/...
~/...$expand=Property/Property2/...


---
In reply to: [280163679](https://github.com/OData/WebApi/pull/1738#discussion_r280163679) [](ancestors = 280163679)